### PR TITLE
refactor tag retrieval and add tag ordering by version

### PR DESCRIPTION
**Pull request by adamdubiel**
Fixes: #146 #149 

Effect of this pull request is adding ordering of tags when there are more than one on a commit. Tags are ordered by highest version. This should fix all problems with multiple tags.

As a side effect, code retrieving tags has been refactored and should be easier to understand and maintain in the future.

Also `tag selector` feature introduced in #141 has been removed - @levsa please let me know if it should be reimplemented, or is current implementation good enough.

### CODE REVIEW

**Comment by adamdubiel** [comment]
@szpak continuing #141 here: no, i meant having multiple tags on one commit

File: src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy [hunk 1]
```diff
@@ -6,50 +6,99 @@ import pl.allegro.tech.build.axion.release.domain.properties.TagProperties
...
+                version      : version,
+                isNextVersion: isVersionNextVersion.get(version),
+                noTagsFound  : versions.isEmpty()
         ]
     }
 
+    private static class VersionComparator implements Comparator<Version> {
+        @Override
+        int compare(Version o1, Version o2) {
+            return -1 * o1.compareTo(o2)
```
Comment (vbuell) [inline src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy]: I believe you may use Collections.reverseOrder() instead.

Comment (adamdubiel) [inline src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionResolver.groovy]: right :)

**Comment by Endron** [comment]
Thank you!

**Pull request merged**